ui/pages/settings/security-tab/delete-metametrics-data-button/delete-metametrics-data-button.test.tsx

```diff
@@ -0,0 +1,207 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fireEvent } from '@testing-library/react';
+import configureStore from '../../../../store/store';
+import { renderWithProvider } from '../../../../../test/lib/render-helpers';
+
+import {
+  getMetaMetricsDataDeletionTimestamp,
+  getMetaMetricsDataDeletionStatus,
+  getMetaMetricsId,
+  getLatestMetricsEventTimestamp,
+} from '../../../../selectors';
+import { openDeleteMetaMetricsDataModal } from '../../../../ducks/app/app';
+import DeleteMetaMetricsDataButton from './delete-metametrics-data-button';
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+  useDispatch: jest.fn(),
+}));
+
+describe('DeleteMetaMetricsDataButton', () => {
+  const useSelectorMock = useSelector as jest.Mock;
+  const useDispatchMock = useDispatch as jest.Mock;
+  const mockDispatch = jest.fn();
+
+  beforeEach(() => {
+    useDispatchMock.mockReturnValue(mockDispatch);
+    useSelectorMock.mockImplementation((selector) => {
+      if (selector === getMetaMetricsId) {
+        return 'fake-metrics-id';
+      }
+      if (selector === getMetaMetricsDataDeletionStatus) {
+        return undefined;
+      }
+      if (selector === getMetaMetricsDataDeletionTimestamp) {
+        return '';
+      }
+
+      return undefined;
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', () => {
+    const store = configureStore({});
+    const { getByText, getByTestId, getAllByText } = renderWithProvider(
+      <DeleteMetaMetricsDataButton />,
+      store,
+    );
+    expect(getByTestId('delete-metametrics-data-button')).toBeInTheDocument();
+    expect(getAllByText('Delete MetaMetrics data')).toHaveLength(2);
+    expect(
```

Comment:
Optional: you can change this to expect(DOM).toMatchInlineSnapshot() and press `u`, so it is less manual to assert the content

Reply:
updater [06c5d65](https://github.com/MetaMask/metamask-extension/pull/24571/commits/06c5d656f6f6ec921f560d8a1e7141ee187ec6b2)